### PR TITLE
Landsat pixel count

### PR DESCRIPTION
**Pull request by NikitaGandhi**
- Initial implementation for USGS landsat pixel counts - counts `total`, `clear` and `clear_<filters>`
- For landsat-8, it also counts `clear_aerosol`

TODO:
- implement tests

### CODE REVIEW

File: libs/stats/odc/stats/_pq_bitmask.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    @property
+    def measurements(self) -> Tuple[str, ...]:
+        band = [
+            "total",
+            "clear",
+            *[f"clear_{r1:d}_{r2:d}" for (r1, r2) in self.filters],
+        ]
+        if self.aerosol_band is not None:
+            band.append("clear_aerosol")
```
Comment (alexgleith) [inline libs/stats/odc/stats/_pq_bitmask.py]:
```suggestion
        measurements = [
            "total",
            "clear",
            *[f"clear_{r1:d}_{r2:d}" for (r1, r2) in self.filters],
        ]
        if self.aerosol_band is not None:
            measurements.append("clear_aerosol")
```

File: libs/stats/odc/stats/_pq_bitmask.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+        return tuple(band)
+
+    def input_data(self, task: Task) -> xr.Dataset:
+        chunks = {"y": -1, "x": -1}
```
Comment (alexgleith) [inline libs/stats/odc/stats/_pq_bitmask.py]: This is done slightly different in the _pq.py file, let's do it like that please :-)

File: libs/stats/odc/stats/_pq_bitmask.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        bands = [self.pq_band]
+        if self.aerosol_band is not None:
+            bands.append(self.aerosol_band)
```
Comment (alexgleith) [inline libs/stats/odc/stats/_pq_bitmask.py:55]: You don't need to do that again...

File: libs/stats/odc/stats/_pq_bitmask.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+        xx = load_with_native_transform(
+            task.datasets,
+            bands=bands,
```
Comment (alexgleith) [inline libs/stats/odc/stats/_pq_bitmask.py]:
```suggestion
            bands=self.measurements,
```

File: libs/stats/odc/stats/_pq_bitmask.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+            geobox=task.geobox,
+            native_transform=self._native_tr,
+            fuser=self._fuser,
+            groupby="solar_day",
+            resampling=self.resampling,
+            chunks=chunks,
+        )
+
+        return xx
```
Comment (alexgleith) [inline libs/stats/odc/stats/_pq_bitmask.py]:
```suggestion
       return load_with_native_transform(
            task.datasets,
            bands=bands,
            geobox=task.geobox,
            native_transform=self._native_tr,
            fuser=self._fuser,
            groupby="solar_day",
            resampling=self.resampling,
            chunks=chunks,
        )
```

**Review by alexgleith — CHANGES_REQUESTED** [review]
A few minor changes.

**Review by uchchwhash — APPROVED** [review]
LGTM!

File: libs/stats/odc/stats/_pq_bitmask.py (hunk 3 above)
Comment (NikitaGandhi) [reply libs/stats/odc/stats/_pq_bitmask.py:55]: This is input bands - `QA_PIXEL` and `SR_QA_AEROSOL` - for measurements

File: libs/stats/odc/stats/_pq_bitmask.py (hunk 2 above)
Comment (NikitaGandhi) [reply libs/stats/odc/stats/_pq_bitmask.py]: It's same as defined in pq file only difference is order of x and y axis

File: libs/stats/odc/stats/_pq_bitmask.py (hunk 5 above)
Comment (NikitaGandhi) [reply libs/stats/odc/stats/_pq_bitmask.py]: done

File: libs/stats/odc/stats/_pq_bitmask.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+class StatsPQLSBitmask(StatsPluginInterface):
+    NAME = "pc_ls_bitmask"
+    SHORT_NAME = NAME
+    VERSION = '0.0.1'
+    PRODUCT_FAMILY = "pixel_quality_statistics"
+
+    def __init__(
+            self,
+            pq_band: str = "QA_PIXEL",
+            aerosol_band: str = None,
```
Comment (alexgleith) [inline libs/stats/odc/stats/_pq_bitmask.py]:
```suggestion
            aerosol_band: Optional[str] = None,
```

File: libs/stats/odc/stats/_pq_bitmask.py (hunk 3 above)
Comment (alexgleith) [reply libs/stats/odc/stats/_pq_bitmask.py:55]: Oh, right sorry!

**Review by alexgleith — APPROVED** [review]
Looks good to me! One little issue with Typing.

File: libs/stats/odc/stats/_pq_bitmask.py (hunk 6 above)
Comment (NikitaGandhi) [reply libs/stats/odc/stats/_pq_bitmask.py]: done

**Review by alexgleith — APPROVED** [review]

**Pull request merged**